src/peft/tuners/lora/layer.py

```diff
@@ -585,14 +585,25 @@ def forward(self, x: torch.Tensor, *args: Any, **kwargs: Any) -> torch.Tensor:
                 if not self.use_dora[active_adapter]:
                     result = result + lora_B(lora_A(dropout(x))) * scaling
                 else:
-                    x = dropout(x)
-                    result = result + self.lora_magnitude_vector[active_adapter](
-                        x,
-                        lora_A=lora_A,
-                        lora_B=lora_B,
-                        scaling=scaling,
-                        base_layer=self.get_base_layer(),
-                    )
+                    if isinstance(dropout, nn.Identity) or not self.training:
+                        result = self.lora_magnitude_vector[active_adapter](
```

Comment:
This line contains a bug, it should be: `result = result + ...`. Further below, `result=None` should be `base_result=None`.

But let's change the whole logic a bit to avoid repeating too much code: In this `if` branch, let's set: `base_result = result`, and in the `else` branch, `base_result=None`. Then just pass `base_result=base_result` to the DoRA layer.